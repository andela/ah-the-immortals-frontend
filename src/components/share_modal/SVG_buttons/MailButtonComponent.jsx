import React, {Component} from 'react';
import PropTypes from 'prop-types';

class MailButtonComponent extends Component {
  static propTypes = {
    article: PropTypes.object.isRequired
  };
  render() {
    const {article: {slug, title}} = this.props;
    const url = process.env.URL;
    const currentUrl  = `${url}/post/${slug}`;
    const shareUrl = `mailto:?subject=${title}&body=${currentUrl}`;
    return (
      <div className="svg">
        <button className="btn btn-link" id="twt" type="button" onClick={() => window.open(shareUrl, '_blank')}>
          <img src="https://img.icons8.com/doodle/64/000000/mail-with-wings.png" alt="avatar" />
        </button>
        <div className="mail-style">Mail</div>
      </div>
    );
  }
}

export default MailButtonComponent;
