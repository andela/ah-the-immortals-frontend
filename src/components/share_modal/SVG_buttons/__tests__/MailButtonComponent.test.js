import { mount } from 'enzyme';
import React from 'react';
import MailButtonComponent from '../MailButtonComponent';


describe('Test for the mail button component', () =>{
  const props = {
    article: {
      slug: ''
    }
  };
  const wrapper = mount(
    <MailButtonComponent {...props} />
  );
  it('Test the mounting of the facebook button', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('Simulates clicking of open button',()=>{
    const openButton = wrapper.find('button').first();
    openButton.simulate('click',
      global.open = jest.fn()
    );
    expect(global.open).toBeCalled();
  });
});
