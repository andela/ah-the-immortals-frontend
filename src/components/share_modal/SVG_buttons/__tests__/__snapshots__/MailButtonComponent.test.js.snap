// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for the mail button component Test the mounting of the facebook button 1`] = `ReactWrapper {}`;
